tests/gui/test_widgets.py

```diff
@@ -2835,6 +2868,56 @@ def test_SpeechBubble__on_download_error(mocker):
     assert bubble.message.text() == error_message
 
 
+def test_CheckMark_eventFilter_hover(mocker):
+    mock_signal = mocker.MagicMock()
+    bubble = SpeechBubble(
+        "mock id",
+        "<b>hello</b>",
+        mock_signal,
+        mock_signal,
+        0,
+        SpeechBubble.MIN_CONTAINER_WIDTH,
+    )
+
+    bubble.check_mark = mocker.MagicMock()
+
+    # Hover enter
+    test_event = QEvent(QEvent.HoverEnter)
+    bubble.eventFilter(bubble, test_event)
+    assert bubble.check_mark.setIcon.call_count
```

Comment:
I think it would be better to be explicit about how many times the icon is expected to be set. (Is seven times okay? I believe probably not.)

It would be even better I think to assert on _which_ file was set as the icon (if practical). Both could be done at once with something in the lines of: `bubble.check_mark.setIcon.assert_called_once_with(some_icon)`.

Note: take this with a grain of salt, I'm not sure how practical it is to specify the icon and I don't think it is worth making the test very complicated in order to check a specific file was used. Checking _how many times_ `setIcon` was called is a good improvement already.